api/src/user/user.utils.ts

```diff
@@ -33,3 +33,16 @@ export function worksheetCoursesToWorksheets(
   }
   return res;
 }
+
+export function getFirstAvailableWsNumber(worksheetNumbers: number[]) {
```

Comment:
If we are returning the worksheets in the order of this number, then they should be numbered in insertion order, so I suggest just an incrementing counter without MEX.

Reply:
This makes sense!